zarr/tests/test_storage.py

```diff
@@ -1,87 +1,96 @@
-import array
```

Comment:
Big diff in this file as I decided to comment everything out and bring it back piece by piece. Here's the current test run:

```
=================================================================================== 13 passed in 0.08s ===================================================================================
❯ pytest zarr/tests/test_storage.py -vvv
================================================================================== test session starts ===================================================================================
platform darwin -- Python 3.11.6, pytest-7.4.3, pluggy-1.3.0 -- /Users/jhamman/miniforge3/envs/zarr-dev/bin/python3.11
cachedir: .pytest_cache
rootdir: /Users/jhamman/Library/CloudStorage/Dropbox/src/zarr-python
configfile: pyproject.toml
plugins: anyio-4.0.0, asyncio-0.23.3
asyncio: mode=Mode.STRICT
collected 13 items                                                                                                                                                                       

zarr/tests/test_storage.py::test_kvstore_repr PASSED                                                                                                                               [  7%]
zarr/tests/test_storage.py::test_ensure_store PASSED                                                                                                                               [ 15%]
zarr/tests/test_storage.py::test_capabilities PASSED                                                                                                                               [ 23%]
zarr/tests/test_storage.py::TestMappingStore::test_get_set_del_contains PASSED                                                                                                     [ 30%]
zarr/tests/test_storage.py::TestMappingStore::test_set_invalid_content PASSED                                                                                                      [ 38%]
zarr/tests/test_storage.py::TestMappingStore::test_writeable_values PASSED                                                                                                         [ 46%]
zarr/tests/test_storage.py::TestMappingStore::test_pickle PASSED                                                                                                                   [ 53%]
zarr/tests/test_storage.py::TestMappingStore::test_hierarchy PASSED                                                                                                                [ 61%]
zarr/tests/test_storage.py::TestDirectoryStore::test_get_set_del_contains PASSED                                                                                                   [ 69%]
zarr/tests/test_storage.py::TestDirectoryStore::test_set_invalid_content PASSED                                                                                                    [ 76%]
zarr/tests/test_storage.py::TestDirectoryStore::test_writeable_values PASSED                                                                                                       [ 84%]
zarr/tests/test_storage.py::TestDirectoryStore::test_pickle PASSED                                                                                                                 [ 92%]
zarr/tests/test_storage.py::TestDirectoryStore::test_hierarchy PASSED  
```